internal/k8s/label.go

```diff
@@ -57,16 +61,22 @@ func injectLabels(entity K8sEntity, labels []model.LabelPair, overwrite bool) (K
 		return K8sEntity{}, err
 	}
 
+	switch obj := entity.Obj.(type) {
+	case *appsv1beta1.Deployment:
+		allowLabelChangesInAppsDeploymentBeta1(obj)
+		applyLabelsToMap(&obj.Spec.Selector.MatchLabels, labels, overwrite)
+	case *appsv1beta2.Deployment:
+		allowLabelChangesInAppsDeploymentBeta2(obj)
+		applyLabelsToMap(&obj.Spec.Selector.MatchLabels, labels, overwrite)
+	case *extv1beta1.Deployment:
+		allowLabelChangesInExtDeploymentBeta1(obj)
+		applyLabelsToMap(&obj.Spec.Selector.MatchLabels, labels, overwrite)
+	case *appsv1.Deployment:
```

Comment:
hmmmm...this is missing a lot of types. Lots of things have Selector.MatchLabels - StatefulSet, ReplicaSet, DaemonSet, etc. cf
https://godoc.org/k8s.io/api/apps/v1

probably simpler to write an extractLabelSelectors function, like we did for extractObjectMetas?